src/java.base/share/classes/java/lang/foreign/Linker.java

```diff
@@ -214,17 +460,35 @@ default MethodHandle downcallHandle(MemorySegment symbol, FunctionDescriptor fun
     }
 
     /**
-     * Creates a method handle which can be used to call a foreign function with the given signature.
-     * The resulting method handle features a prefix parameter (as the first parameter) corresponding to the foreign function
-     * entry point, of type {@link MemorySegment}, which is used to specify the address of the target function
-     * to be called.
+     * Creates a method handle which is used to call a foreign function with the given signature.
+     * <p>
+     * The Java {@linkplain java.lang.invoke.MethodType method type} associated with the returned method handle is
+     * {@linkplain FunctionDescriptor#toMethodType() derived} from the argument and return layouts in the function descriptor,
+     * but features an additional leading parameter of type {@link MemorySegment}, from which the address of the target
+     * foreign function is derived. Moreover, if the function descriptor's return layout is a group layout, the resulting
+     * downcall method handle accepts an additional leading parameter of type {@link SegmentAllocator}, which is used by
+     * the linker runtime to allocate the memory region associated with the struct returned by the downcall method handle.
+     * <p>
+     * Upon invoking a downcall method handle, the linker runtime will guarantee the following for any argument
+     * {@code A} of type {@link MemorySegment} whose corresponding layout is an {@linkplain AddressLayout address layout}:
+     * <ul>
+     *     <li>{@code A.scope().isAlive() == true}. Otherwise, the invocation throws {@link IllegalStateException};</li>
+     *     <li>The invocation occurs in a thread {@code T} such that {@code A.isAccessibleBy(T) == true}.
+     *     Otherwise, the invocation throws {@link WrongThreadException}; and</li>
+     *     <li>{@code A} is kept alive during the invocation. For instance, if {@code A} has been obtained using a
+     *     {@linkplain Arena#ofConfined() confined arena}, any attempt to {@linkplain Arena#close() close}
```

Comment:
Is that correct? Do you mean a shared arena?

Reply:
The text is correct (you can still close a confined arena from inside an upcall), but I agree that perhaps using a shared arena might be a bit more intuitive.